File: Lab4/main/java/window/controller/CommandFactory.java
package window.controller;

import window.model.Color;

import java.util.List;

public class CommandFactory {
    public static Command getCommand(String com, List<Object> list) {
        Receiver rec = (Receiver) list.get(0);
        if (com.startsWith("close")) return new CloseCommand(rec);
        else if (com.startsWith("open")) return new OpenCommand(rec);
        else if (com.startsWith("counter")) return new CounterCommand(rec, (String) list.get(1));
        else if (com.startsWith("proxy")) return new ProxyCommand(rec, (Color) list.get(1));
        return null;
    }
}

File: Lab6/src/main/kotlin/controller/CommandFactory.kt
package controller

import java.lang.IllegalArgumentException

object CommandFactory {
    fun getCommand(com: String, list: List<Any?>): Command {
        val rec: Receiver = list[0] as Receiver
        return when (com){
            "close" -> CloseCommand(rec)
            "open" -> OpenCommand(rec)
            "counter" -> CounterCommand(rec, list[1] as String)
            else -> throw IllegalArgumentException()
        }
    }
}